File: easy/4/4.erl
% c('4.erl').
% '4'.start().

-module('4').
-compile(export_all).

start()->
    io:format("Enter length of password"),
    Length = hd(string:tokens(io:get_line("> "),"\r\n")),
password(list_to_integer(Length)).

password(0)->
    [];

password(Length)->
    RandomInt = 33 + random:uniform(127 - 33),
    Res = [RandomInt| password(Length-1)],
    Res;

password() ->
    password(8).

File: easy/4/prolog/4.pl
%% ['4.pl'].
%% start.

:- use_module(library(random)).

start(Pwd):-
    format("Enter length of password ~n"),
    read(Length),
    gen_password(Length, Pwd).

gen_password(0, []).

gen_password(Length, [X|Y]):-
    random(33, 127, X),
    Length1 is Length -1,
    gen_password(Length1, Y).